samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -26,7 +26,7 @@
 
 public abstract class AbstractApplicationRunner implements ApplicationRunner {
```

Comment:
Yes the assumption is right. 

Abstract classes and interfaces are not mutually exclusive. For example see:
http://docs.oracle.com/javase/8/docs/api/java/util/AbstractMap.html

However, I will take a second look at this case tomorrow, to ensure that the interface is still useful. If not, I'll open a separate PR to remove it. 

Excellent point about the method name. streamFromConfig was the original name when the method took config as a parameter. I failed to rename it after the refactor. Will fix. Thanks!